src/app/(dashboard)/attestation/attestationModal.tsx

```diff
@@ -64,6 +71,16 @@ export function GenerateAttestationModal({
     }
   }, [chainId]);
 
+  useEffect(() => {
+    if (creatPrivateAttestation != null) {
+      if (creatPrivateAttestation) {
+        handleFetchAttestationPrivateData();
+      } else {
+        handleFetchAttestationPublicData();
+      }
+    }
+  }, [creatPrivateAttestation]);
```

Comment:
This makes the complexity of the code for a newcomer a bit harder to understand,

instead of having the flag as a boolean, and use it with a third argument as "null", I think you should use this flag as a string, and the string content would be "public" or "private", makes the logic flow much easier to understand and to follow